Simplenote/build.gradle

```diff
@@ -15,6 +15,13 @@ android {
             shrinkResources false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
+
+        // Dedicated build type for screenshots so that we can add the special
+        // permissions only to it, keeping debug and release closer together to
+        // make bug hunting easier.
+        screenshots {
+            initWith debug
```

Comment:
@theck13 you suggested to use `release`, but doing so would have resulted in more configuration in order to assemble the APK for the testing.

```
$ ./gradlew assembleScreenshots

> Configure project :Simplenote
WARNING: API 'variantOutput.getProcessManifest()' is obsolete and has been replaced with 'variantOutput.getProcessManifestProvider()'.
It will be removed at the end of 2019.
For more information, see https://d.android.com/r/tools/task-configuration-avoidance.
To determine what is calling variantOutput.getProcessManifest(), use -Pandroid.debug.obsoleteApi=true on the command line to display more information.

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':Simplenote:handleScreenshotsMicroApk'.
> Could not resolve all task dependencies for configuration ':Simplenote:screenshotsWearBundling'.
   > Could not resolve project :Wear.
     Required by:
         project :Simplenote
      > Unable to find a matching variant of project :Wear:
          - Variant 'debugApiElements' capability snandroid:Wear:unspecified:
              - Incompatible attributes:
                  - Required com.android.build.api.attributes.BuildTypeAttr 'screenshots' and found incompatible value 'debug'.
                  - Required org.gradle.usage 'java-runtime' and found incompatible value 'java-api'.
              - Other attributes:
                  - Found com.android.build.api.attributes.VariantAttr 'debug' but wasn't required.
                  - Required com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Apk' and found compatible value 'Apk'.
          - Variant 'debugBundleElements' capability snandroid:Wear:unspecified:
              - Incompatible attribute:
                  - Required org.gradle.usage 'java-runtime' and found incompatible value 'android-bundle'.
              - Other attributes:
                  - Required com.android.build.api.attributes.BuildTypeAttr 'screenshots' but no value provided.
                  - Found com.android.build.api.attributes.VariantAttr 'debug' but wasn't required.
                  - Required com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Apk' but no value provided.
          - Variant 'debugMetadataElements' capability snandroid:Wear:unspecified:
              - Incompatible attributes:
                  - Required com.android.build.api.attributes.BuildTypeAttr 'screenshots' and found incompatible value 'debug'.
                  - Required com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Apk' and found incompatible value 'Metadata'.
              - Other attributes:
                  - Found com.android.build.api.attributes.VariantAttr 'debug' but wasn't required.
                  - Required org.gradle.usage 'java-runtime' but no value provided.
          - Variant 'debugRuntimeElements' capability snandroid:Wear:unspecified:
              - Incompatible attribute:
                  - Required com.android.build.api.attributes.BuildTypeAttr 'screenshots' and found incompatible value 'debug'.
              - Other attributes:
                  - Found com.android.build.api.attributes.VariantAttr 'debug' but wasn't required.
                  - Required com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Apk' and found compatible value 'Apk'.
                  - Required org.gradle.usage 'java-runtime' and found compatible value 'java-runtime'.
          - Variant 'releaseApiElements' capability snandroid:Wear:unspecified:
              - Incompatible attributes:
                  - Required com.android.build.api.attributes.BuildTypeAttr 'screenshots' and found incompatible value 'release'.
                  - Required org.gradle.usage 'java-runtime' and found incompatible value 'java-api'.
              - Other attributes:
                  - Found com.android.build.api.attributes.VariantAttr 'release' but wasn't required.
                  - Required com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Apk' and found compatible value 'Apk'.
          - Variant 'releaseBundleElements' capability snandroid:Wear:unspecified:
              - Incompatible attribute:
                  - Required org.gradle.usage 'java-runtime' and found incompatible value 'android-bundle'.
              - Other attributes:
                  - Required com.android.build.api.attributes.BuildTypeAttr 'screenshots' but no value provided.
                  - Found com.android.build.api.attributes.VariantAttr 'release' but wasn't required.
                  - Required com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Apk' but no value provided.
          - Variant 'releaseMetadataElements' capability snandroid:Wear:unspecified:
              - Incompatible attributes:
                  - Required com.android.build.api.attributes.BuildTypeAttr 'screenshots' and found incompatible value 'release'.
                  - Required com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Apk' and found incompatible value 'Metadata'.
              - Other attributes:
                  - Found com.android.build.api.attributes.VariantAttr 'release' but wasn't required.
                  - Required org.gradle.usage 'java-runtime' but no value provided.
          - Variant 'releaseRuntimeElements' capability snandroid:Wear:unspecified:
              - Incompatible attribute:
                  - Required com.android.build.api.attributes.BuildTypeAttr 'screenshots' and found incompatible value 'release'.
              - Other attributes:
                  - Found com.android.build.api.attributes.VariantAttr 'release' but wasn't required.
                  - Required com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Apk' and found compatible value 'Apk'.
                  - Required org.gradle.usage 'java-runtime' and found compatible value 'java-runtime'.
```

This setup works, and it kinda makes sense to me to use the debug one, since we need to generate both app and tests APKs for `screengrab` and the only test APK I can generate is debug.

What do you think?

Reply:
If it makes the configuration easier, that's fine with me.  One thing to note is that if `screenshot` uses `initWith debug`, it will use the `.debug` suffix on the package and overwrite any exiting `debug` build.  I don't think that will cause any major problems other than requiring uninstall before install in cases when running one variant is behind the currently installed variant due to branching.